docs/docs/cmd/planner/task/task-set.md

```diff
@@ -0,0 +1,91 @@
+# planner task set
+
+Updates a Microsoft Planner task
+
+## Usage
+
+```sh
+m365 planner task set [options]
+```
+
+## Options
+
+`-i, --id <id>`
+: ID of the task.
+
+`-t, --title [title]`
+: Title of the task to add.
+
+`--bucketId [bucketId]`
+: Bucket ID to which the task belongs. Specify either `bucketId` or `bucketName` but not both.
+
+`--bucketName [bucketName]`
+: Bucket Name to which the task belongs. The bucket needs to exist in the selected plan. Specify either `bucketId` or `bucketName` but not both.
+
+`--planId [planId]`
+: Plan ID to which the task belongs. Specify either `planId` or `planName` but not both.
+
+`--planName [planName]`
+: Plan Name to which the task belongs. Specify either `planId` or `planName` but not both.
+
+`--ownerGroupId [ownerGroupId]`
+: ID of the group to which the plan belongs. Specify `ownerGroupId` or `ownerGroupName` when using `planName`.
+
+`--ownerGroupName [ownerGroupName]`
+: Name of the group to which the plan belongs. Specify `ownerGroupId` or `ownerGroupName` when using `planName`.
+
+`--startDateTime [startDateTime]`
+: The date and time when the task started. This should be defined as a valid ISO 8601 string. `2021-12-16T18:28:48.6964197Z`
+
+`--dueDateTime [dueDateTime]`
+: The date and time when the task is due. This should be defined as a valid ISO 8601 string. `2021-12-16T18:28:48.6964197Z`
+
+`--percentComplete [percentComplete]`
+: Percentage of task completion. Number between 0 and 100.
+  - When set to 0, the task is considered _Not started_. 
+  - When set between 1 and 99, the task is considered _In progress_.
+  - When set to 100, the task is considered _Completed_.
+
+`--assignedToUserIds [assignedToUserIds]`
+: The comma-separated IDs of the assignees the task is assigned to. Specify either `assignedToUserIds` or `assignedToUserNames` but not both.
+
+`--assignedToUserNames [assignedToUserNames]`
+: The comma-separated UPNs of the assignees the task is assigned to. Specify either `assignedToUserIds` or `assignedToUserNames` but not both.
+
+`--description [description]`
+: Description of the task
+
+`--orderHint [orderHint]`
+: Hint used to order items of this type in a list view. The format is defined as outlined [here](https://docs.microsoft.com/en-us/graph/api/resources/planner-order-hint-format?view=graph-rest-1.0).
+
+`--assigneePriority [assigneePriority]`
+: Hint used to order items of this type in a list view. The format is defined as outlined [here](https://docs.microsoft.com/en-us/graph/api/resources/planner-order-hint-format?view=graph-rest-1.0).
+
+`--conversationThreadId [conversationThreadId]`
+: Thread id of the conversation on the task. This is the id of the conversation thread object created in the group.
```

Comment:
I have changed the description of the option so that it is a bit more informative. But when applying this option, visually not much happens to the task. In the backend you can see that the new ID is linked but visually the option to leave a comment breaks. Or it resets the conversation. Currently I have left the option in but don't know if this is the best idea. What do you think about this?

Reply:
Thanks! What value would you specify in there? How to get this ID? If all of that is too complicated/unclear, we could also decide to leave this option out for now and add it later when we have a better story for it.

Reply:
You can get a list of threads for example by using [this graph endpoint](https://docs.microsoft.com/en-us/graph/api/group-list-threads?view=graph-rest-1.0&tabs=http). When testing I used this Id to change the conversation thread of my task `AAQkAGE0MTczMzEyLTI5M2EtNDE3ZC04ZjExLWNlYzI5ODFkNjlkZAMkABAAKFVlCe6axUuI4yNU-8hchBAAKFVlCe6axUuI4yNU-8hchB==`. Personally, I think it's quite complex. Mainly because I don't have a lot of experience with conversation threads yet. So I think it won't be a bad idea to leave it out for now.

Reply:
That said, is the PR ready for review or do you still need to remove this option and the accompanying code?

Reply:
Now it is ready for review 😄